internal/broker/broker.go

```diff
@@ -441,18 +442,20 @@ func (b *Broker) handleIsAuthenticated(ctx context.Context, session *sessionInfo
 		defer cancel()
 		t, err := session.authCfg.oauth.DeviceAccessToken(expiryCtx, response, b.providerInfo.AuthOptions()...)
 		if err != nil {
-			return AuthRetry, errorMessage{Message: fmt.Sprintf("could not authenticate user: %v", err)}
+			slog.Error(err.Error())
+			return AuthRetry, errorMessage{Message: "could not authenticate user remotely"}
 		}
 
 		rawIDToken, ok := t.Extra("id_token").(string)
 		if !ok {
-			return AuthDenied, errorMessage{Message: "could not get id_token"}
+			return AuthDenied, errorMessage{Message: "could not get ID token"}
```

Comment:
The only place I could find where this error message is printed is in the PAM module, whose output doesn't end up in the logs IIUC (but is directly shown to the user). So how about we log any error message we return in this function, to ensure that it's in the logs?

Reply:
I don't think it's necessary to log everything here. This is a simple error message that does not contain any useful/sensitive info, so I think it's fine to keep it as is.

Reply:
I'm not talking about sensitive or additional info, but don't you think it makes sense to be able to tell from the logs that it's this place in the code where the authentication failed?

If I make the broker return this error message, by applying this patch:

```
diff --git a/internal/broker/broker.go b/internal/broker/broker.go
index 288eda0..4d46805 100644
--- a/internal/broker/broker.go
+++ b/internal/broker/broker.go
@@ -447,7 +447,7 @@ func (b *Broker) handleIsAuthenticated(ctx context.Context, session *sessionInfo
                }
 
                rawIDToken, ok := t.Extra("id_token").(string)
-               if !ok {
+               if true {
                        return AuthDenied, errorMessage{Message: "could not get ID token"}
                }
```

And I do the device authentication via `sudo login`, then "could not get ID token" is printed in the terminal, but there is no message about this authentication failure in the logs (unless debug logging is enabled for the PAM module). When trying to log in via GDM, it doesn't even show the error message in GDM. That's probably a bug, but even if it did, I think it would be helpful for debugging if we could tell from the logs why the authentication failed. What do you think?